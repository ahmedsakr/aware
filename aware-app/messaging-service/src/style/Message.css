.message {
    margin-top: 10px;
    display: flex;
}

.message-profile {
    border: 2px solid rgb(54, 56, 58);
    border-radius: 5px;
    text-align: center;
    padding: 0px;
    margin-right: 5px;
}

.message-profile-img {
    max-width: 30%;
}

.message-profile-name {
    background-color: #bac1b8;
    text-align: center;
    margin: 0px;
}

.message-text {
    background-color: #a1c6ea;
    border-radius: 10px;
    padding: 5px;
    margin-right: 5px;
}

.message-timestamp {
    padding: 0px;
    text-align: left;
    font-size: 0.6em;
}