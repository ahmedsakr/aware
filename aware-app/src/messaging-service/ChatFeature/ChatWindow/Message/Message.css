.received-message {
    margin-top: 10px;
    display: flex;
}

.received-message-profile {
    border: 2px solid rgb(54, 56, 58);
    border-radius: 5px;
    text-align: center;
    padding: 0px;
    margin-right: 5px;
}

.received-message-profile-img {
    max-width: 30%;
}

.received-message-profile-name {
    background-color: #bac1b8;
    text-align: center;
    margin: 0px;
}

.received-message-text {
    background-color: #a1c6ea;
    border-radius: 10px;
    padding: 5px;
    margin-right: 5px;
}

.received-message-timestamp {
    padding: 0px;
    text-align: left;
    font-size: 0.6em;
}

.sent-message {
    margin-top: 10px;
    display: flex;
    margin-right: 5px;
    padding-right: 15px;
    text-align: center;
}

.sent-message-profile {
    padding: 0px;
    font-size: 0.8em;
}

.sent-message-profile-name {
    margin: 0px;
}

.sent-message-text {
    display: flex;
    align-items: flex-end;
    text-align: left;
}

.sent-message-text > p {
    background-color: #507dbc;
    border-radius: 10px;
    padding: 5px;
    margin-left: 5px;
    width: 100%;
    margin-bottom: 0px;
}

.sent-message-timestamp {
    padding: 0px;
    text-align: right;
    font-size: 1em;
    display: flex;
    align-items: flex-end;
    justify-content: flex-end;
    opacity: 0.8;
}

.sent-message-timestamp > p {
    margin: 0;
    padding-right: 5px;
}