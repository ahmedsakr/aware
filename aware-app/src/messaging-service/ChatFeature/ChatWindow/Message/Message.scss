/*
 * Message styling
 *
 * Defines layout fundamentals for every message.
 */
$received_message_background_color: #a1c6ea;
$sent_message_background_color: #507dbc;

.message {
    margin-top: 10px;
    display: flex;
}

.received-message {
    margin-left: 5px;
    padding-left: 15px;
}

.sent-message {
    margin-right: 5px;
    padding-right: 15px;
}

/*
   * Message text styling
   *
   * The messages bubbles are designed to be fluid in size relative to the size of the text.
   */

.message-text {
    display: flex;
    align-items: flex-start;
    text-align: left;
}

.sent-message-text {
    justify-content: right;
}

.received-message-text {
    justify-content: left;
}

.message-text>p {
    border-radius: 10px;
    margin-bottom: 0px;
    padding: 0px 10px 0px 10px;
}

.received-message-text>p {
    background-color: $received_message_background_color;
    margin-right: 5px;
}

.sent-message-text>p {
    background-color: $sent_message_background_color;
    margin-left: 5px;
}

/*
   * Profile styling
   * 
   * Profile elements for every message include the profile picture and the first
   * name of the sender. The name is centered under the profile image.
   */

.message-profile {
    padding: 0px;
    font-size: 0.8em;
    text-align: center;
}

.message-profile-name {
    margin: 0px;
}