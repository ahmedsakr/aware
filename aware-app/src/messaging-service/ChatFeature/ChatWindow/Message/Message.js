import React, { Component } from 'react';
import './Message.css'
import ProfilePicture from '../../Profile/ProfilePicture'

class Message extends Component {

    produceMessageByThem(content) {
        return (
            <div class="col-sm-12 message received-message">
                <div class="col-sm-1 message-profile">
                    <ProfilePicture instance="message" activity="online"  picture={content.avatar}/>
                    <p class="message-profile-name">{content.studentName}</p>
                </div>

                <div class="col-sm-4 message-text received-message-text">
                    <p>{content.text}</p>
                </div>

                <div class="col-sm-2 message-timestamp received-message-timestamp">
                    <p>{content.timestamp}</p>
                </div>
            </div>
        );
    }

    produceMessageByMe(content) {
        return (
            <div class="col-sm-12 message sent-message">
                <div class="offset-sm-5 col-sm-2 message-timestamp sent-message-timestamp">
                    <p>{content.timestamp}</p> 
                </div>
                <div class="col-sm-4 message-text sent-message-text">
                    <p>{content.text}</p>
                </div>
                <div class="col-sm-1 message-profile">
                    <ProfilePicture instance="message" activity="online"  picture={content.avatar}/>
                    <p class="message-profile-name">{content.studentName}</p>
                </div>
            </div>
        );
    }
    
    render() {
        if (this.props.content.studentName == this.props.name) {
            return this.produceMessageByMe(this.props.content);
        } else {
            return this.produceMessageByThem(this.props.content);
        }
    }
}

export default Message;
