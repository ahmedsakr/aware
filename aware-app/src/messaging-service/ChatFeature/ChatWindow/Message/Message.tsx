import React from 'react';
import './Message.css'
import ProfilePicture from '../../Profile/ProfilePicture'

export interface UserMessageContents {
    username: string,
    avatar: string,
    content: string,
    timestamp: string,
};

type MessageProps = {
    content: UserMessageContents,
    name: string
};

type MessageState = { };

export default class Message extends React.Component<MessageProps, MessageState> {

    produceMessageByThem(content: UserMessageContents): JSX.Element {
        return (
            <div className="col-sm-12 message received-message">
                <div className="col-sm-1 message-profile">
                    <ProfilePicture instance="message" activity="online"  picture={content.avatar}/>
                    <p className="message-profile-name">{content.username}</p>
                </div>

                <div className="col-sm-4 message-text received-message-text">
                    <p>{content.content}</p>
                </div>

                <div className="col-sm-2 message-timestamp received-message-timestamp">
                    <p>{content.timestamp}</p>
                </div>
            </div>
        );
    }

    produceMessageByMe(content: UserMessageContents): JSX.Element {
        return (
            <div className="col-sm-12 message sent-message">
                <div className="offset-sm-5 col-sm-2 message-timestamp sent-message-timestamp">
                    <p>{content.timestamp}</p> 
                </div>
                <div className="col-sm-4 message-text sent-message-text">
                    <p>{content.content}</p>
                </div>
                <div className="col-sm-1 message-profile">
                    <ProfilePicture instance="message" activity="online"  picture={content.avatar}/>
                    <p className="message-profile-name">{content.username}</p>
                </div>
            </div>
        );
    }
    
    render() {
        if (this.props.content.username === this.props.name) {
            return this.produceMessageByMe(this.props.content);
        } else {
            return this.produceMessageByThem(this.props.content);
        }
    }
}
