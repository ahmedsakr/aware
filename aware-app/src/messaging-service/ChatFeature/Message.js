import React, { Component } from 'react';
import './Message.css';

class Message extends Component {

    render() {
        const { student, avatar, message, timestamp } = this.props;

        return (
            <div class="message">
                <div class="col-sm-2 message-profile">
                    <img class="message-profile-img" src={process.env.PUBLIC_URL + avatar} alt={student} />
                    <p class="message-profile-name">{student}</p>
                </div>

                <div class="col-sm-7 message-text">
                    {message}
                </div>

                <div class="col-sm-3 message-timestamp">
                    {timestamp}
                </div>
            </div>
        );
    }
}

export default Message;
