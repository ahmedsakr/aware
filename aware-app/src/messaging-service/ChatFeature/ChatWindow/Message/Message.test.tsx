import React from 'react';
import { mount } from 'enzyme';

import Message, { UserMessageContents } from './Message';

describe('Message', () => {
    it('should render component correctly', () => {

        let props: UserMessageContents = {
            username: "Ahmed",
            content: "Yeah, I am kinda drunk.",
            timestamp: "December 1, 2018 - 12:22 pm",
            avatar: "/icons8-user-80blue.png"
        }

        const wrapper = mount(
            <Message
                name="Ahmed"
                content={props} />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
