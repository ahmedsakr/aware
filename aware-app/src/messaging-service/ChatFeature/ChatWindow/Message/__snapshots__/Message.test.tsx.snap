version https://git-lfs.github.com/spec/v1
oid sha256:bf1240fa9a57abb79f5ba7a82a7a32b077add668e01803e7465179809631b169
size 1506
